Update contributing.rst for Windows users

coriegulik opened the pull request:
<!--
Thanks for contributing to JupyterLab!
Please fill out the following items to submit a pull request.
See the contributing guidelines for more information:
https://github.com/jupyterlab/jupyterlab/blob/main/CONTRIBUTING.md
-->

## References

<!-- Note issue numbers this pull request addresses (should be at least one, see contributing guidelines above). -->

<!-- Note any other pull requests that address this issue and how this pull request is different. -->
none

## Code changes

<!-- Describe the code changes and how they address the issue. -->
Added instructions for Windows users on "Setting up a local development environment"
## User-facing changes
need instructions for Windows users on "Setting up a local development environment" 
<!-- Describe any visual or user interaction changes and how they address the issue. -->

<!-- For visual changes, include before and after screenshots or GIF/mp4/other video demo here. -->

## Backwards-incompatible changes
none
<!-- Describe any backwards-incompatible changes to JupyterLab public APIs. -->

fcollonval commented on docs/source/developer/contributing.rst:
```diff
@@ -288,7 +288,7 @@ about 7 minutes again.
 ------------------------------------------
 This section explains how to set up a local development environment. We assume you use GNU/Linux,
-macOS, or Windows Subsystem for Linux.
+macOS, or Windows Subsystem for Linux. If using Windows, we recommend installing Anaconda for windows and then using the Anaconda command prompt for all installation steps.
```
We could add a link to ease discovery

```suggestion
macOS, or Windows Subsystem for Linux. If using Windows, we recommend installing `Anaconda for windows <https://www.anaconda.com/download>`__ and then using the Anaconda command prompt for all installation steps.
```

fcollonval reviewed (commented):
Thanks @coriegulik 

I have a minor suggestion.

coriegulik commented:
Thank you for the suggestion. 
I deleted the head repository because I received error messages and was unable to resolve them. I have readded, implemented your suggestion, and submitted a new PR (https://github.com/jupyterlab/jupyterlab/pull/14665)
